src/richie/apps/courses/templates/courses/cms/course_detail.html

```diff
@@ -368,22 +368,28 @@ <h2 class="course-detail__title">{% blocktrans context "course_detail__title" %}
             {% block runs %}
                 <div class="course-detail__aside">
                     {% render_model_add course "" "" "get_admin_url_to_add_run" %}
-                    {% with runs_dict=current_page.course.course_runs_dict %}
-                        {% with open_visible_runs=runs_dict.0|add:runs_dict.1|add:runs_dict.2|visible_on_course_page:request.toolbar.edit_mode_active %}
-                            <!--
-                                As course runs are rendered through React, we use meta tags to
-                                setup course runs RFDa properties.
-                             -->
-                            {% for run in open_visible_runs %}
-                                <span rel="hasCourseInstance" typeof="CourseInstance">
-                                    <meta property="name" content="{{ run.title }}"  />
-                                    <meta property="inLanguage" content="{{ run.get_languages_display }}"  />
-                                    <meta property="courseMode" content="online"  />
-                                    <meta property="startDate" content="{{ run.start|date:'Y-m-d' }}"  />
-                                    <meta property="endDate" content="{{ run.end|date:'Y-m-d' }}"  />
+                    {% with runs=current_page.course.course_runs|visible_on_course_page:request.toolbar.edit_mode_active %}
+                        <!--
+                            As course runs are rendered through React, we use meta tags to
+                            setup course runs RFDa properties.
+                            -->
+                        <span rel="offers" typeof="Offer">
+                            <meta property="category" content="Partially Free" />
+                        </span>
+                        {% for run in runs %}
+                            <span rel="hasCourseInstance" typeof="CourseInstance">
+                                <meta property="name" content='{{ run.title|default:current_page.get_title }}' />
+                                <meta property="inLanguage" content="{{ run.get_languages_display }}" />
+                                <meta property="courseMode" content="online" />
+                                <span property="courseSchedule" typeof="Schedule">
+                                    <meta property="startDate" content="{{ run.start|date:'Y-m-d' }}" />
+                                    <meta property="endDate" content="{{ run.end|date:'Y-m-d' }}" />
+                                    <meta property="duration" content="{{ current_page.course.pt_effort }}" />
+                                    <meta property="repeatCount" content="1" />
+                                    <meta property="repeatFrequency" content="Weekly" />
```

Comment:
Removed `courseSchedule` and replaced by `courseWorkload`.

Ref: https://developers.google.com/search/docs/appearance/structured-data/course-info#guidelines
> Only one of courseSchedule or courseWorkload properties is required. We recommend the courseSchedule property if you want to highlight pacing (for example, 4 hours per week), or start and end dates. Otherwise, courseWorkload works for most cases.

Reply:
I think it could be feasible to fulfill `repeatCount` and `repeatFrequency` dynamically, but only for instructor pace courses. For self paced, I think it only makes sense to fulfill the `courseWorkload`. So to reduce the effort of this PR, I think we just need to fulfill the `courseWorkload` that works for both cases.

Reply:
Ok right!